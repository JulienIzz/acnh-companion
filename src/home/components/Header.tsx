import {StyleSheet, Text, View, ImageBackground} from 'react-native';
import React from 'react';

const HEADER_HEIGHT = 60;
const HEADER_BOTTOM_MARGIN = 20;
const HEADER_FONT_SIZE = 30;

export const Header = ({
  headerText,
  headerImagePath,
}: {
  headerText: string;
  headerImagePath: any;
}) => {
  return (
    <View style={styles.container}>
      <ImageBackground style={styles.background} source={headerImagePath}>
        <Text style={styles.text}>{headerText}</Text>
      </ImageBackground>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: HEADER_HEIGHT,
    marginBottom: HEADER_BOTTOM_MARGIN,
  },
  text: {
    fontSize: HEADER_FONT_SIZE,
    fontWeight: 'bold',
  },
  background: {
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center',
  },
});
