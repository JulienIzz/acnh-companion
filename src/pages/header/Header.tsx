import {StyleSheet, Text, View, ImageBackground} from 'react-native';
import React from 'react';
import Typography from '../../theme/typography/Typography';

const HEADER_HEIGHT = 60;
const HEADER_BOTTOM_MARGIN = 40;

interface HeaderProps {
  headerText: string;
  headerImagePath: any;
}

export const Header = ({headerText, headerImagePath}: HeaderProps) => {
  return (
    <View style={styles.container}>
      <ImageBackground style={styles.background} source={headerImagePath}>
        <Typography variant="header" text={headerText} />
      </ImageBackground>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: HEADER_HEIGHT,
    marginBottom: HEADER_BOTTOM_MARGIN,
  },
  background: {
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center',
  },
});
