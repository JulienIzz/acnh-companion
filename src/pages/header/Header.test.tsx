import {Header} from './Header';
import React from 'react';
import {render, screen} from '@testing-library/react-native';

test('header with some text', async () => {
  // PARTIE 1 : On rend le composant avec les constantes qu'on veut
  const expectedHeaderTextForTest = 'Mock Text Header';
  const expectedHeaderImageForTest = require('./img/header_background.jpg');

  render(
    <Header
      headerText={expectedHeaderTextForTest}
      headerImagePath={expectedHeaderImageForTest}
    />,
  );

  // PARTIE 2 : On cherche les données qu'on veut vérifier dans le composant rendu / On vérifie les comportements
  const HeaderOutput = screen.getByText(expectedHeaderTextForTest);

  // PARTIE 3 : On fait des vérifications
  expect(HeaderOutput).toBeTruthy();
  expect(screen.toJSON()).toMatchSnapshot();
});
