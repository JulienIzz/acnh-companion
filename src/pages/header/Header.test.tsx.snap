// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header with some text 1`] = `
<View
  style={
    {
      "height": 60,
      "marginBottom": 5,
      "width": "100%",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/pages/header/img/header_background.jpg",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": "100%",
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <Text
      style={
        {
          "color": "black",
          "fontSize": 30,
          "fontWeight": "bold",
        }
      }
    >
      Mock Text Header
    </Text>
  </View>
</View>
`;
